cpp/oneapi/dal/backend/linalg/matrix.hpp

```diff
@@ -33,6 +27,10 @@ enum class layout {
     column_major,
 };
 
+inline constexpr layout transpose_layout(layout l) {
+    return (l == layout::row_major) ? layout::column_major : layout::row_major;
```

Comment:
Why you cannot use the existing `data_layout` enum and created another one? You expect they can differ in future?

Reply:
There is `unknown` layout in oneDAL. I don't want to handle these cases in linalg module. I suppose the linalg layout will not be extended.